projects/packages/my-jetpack/_inc/hooks/use-products/index.js

```diff
@@ -0,0 +1,54 @@
+/* global myJetpackInitialState */
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_ID } from '../../state/store';
+import useMyJetpackConnection from '../use-my-jetpack-connection';
+
+/**
+ * React custom hook to access some products information.
+ *
+ * @returns {Object} productsThatRequiresUserConnection, productsWithActivePlugin, disconnectedAndOnePluginInstalled.
+ */
+export default function useProducts() {
+	const { isSiteConnected } = useMyJetpackConnection();
+
+	const productsThatRequiresUserConnection = useSelect( select =>
+		select( STORE_ID ).getProductsThatRequiresUserConnection()
+	);
+
+	const productsWithActivePlugin = getProductsWithActivePlugin();
+
+	const disconnectedAndOnePluginInstalled =
+		! isSiteConnected && productsWithActivePlugin.length === 1;
+
+	return {
+		productsThatRequiresUserConnection,
+		productsWithActivePlugin,
+		disconnectedAndOnePluginInstalled,
```

Comment:
I wonder whether worths computing this `disconnectedAndOnePluginInstalled` here in the hook, because maybe it could be implemented straight to the component. Just thinking loud.